src/lib.rs

```diff
@@ -436,6 +438,7 @@ pub struct Adapter {
 pub struct Device {
     context: Arc<C>,
     id: <C as Context>::DeviceId,
+    error_sink: ErrorSink
```

Comment:
It seems to me that `ErrorSink` is only needed in the direct backend, right?
We could make it a part of the direct objects that have methods. I.e. in the backend, instead of this:
```rust
type BufferId = wgc::id::BufferId;
```
We could have:
```rust
struct Buffer {
  id: wgc::id::BufferId,
  error_sink: ErrorSink,
}
```
I do wonder how verbose it would be comparing to the current code here, but at least it's going to be properly concealed in the backend.

Reply:
Ah, so direct `impl Context` could just have `type BufferId = Buffer` instead of `type BufferId = wgc::id::BufferId`? That could work, then it would be contained in the right place within the direct backend, that would make sense!

Yeah, the wrapping structs feel a bit annoyingly verbose, but if you are ok with it, I'll try with that

Reply:
hopefully, it's not for too many of them, and not too annoying?